src/Compilers/CSharp/Portable/BoundTree/BoundNodes.xml

```diff
@@ -2083,6 +2109,21 @@
     <Field Name="IsExplicitNotNullTest" Type="bool"/>
   </Node>
 
+  <Node Name="BoundListPatternClause" Base="BoundNode">
+    <Field Name="ElementType" Type="TypeSymbol"/>
+    <Field Name="Subpatterns" Type="ImmutableArray&lt;BoundPattern&gt;" Null="disallow"/>
+    <Field Name="HasSlice" Type="bool"/>
+    <!-- This will be null if input type is an array -->
+    <Field Name="IndexerInfo" Type="IndexerArgumentInfo?" />
+  </Node>
+
+  <Node Name="BoundSlicePattern" Base="BoundPattern">
+    <Field Name="SliceType" Type="TypeSymbol"/>
```

Comment:
Do we need `BoundSlicePattern.SliceType` field?
`BoundSlicePattern.Pattern` should already store this as its input type.

Same question for `BoundListPatternClause.ElementType`. Each of the sub-patterns already stores an input type.